src/System.Runtime/tests/System/GC.cs

```diff
@@ -226,4 +226,85 @@ private class TestObject
             }
         }
     }
-}
\ No newline at end of file
+
+    [Fact]
+    public static void ReRegisterForFinalize()
+    {
+        ReRegisterForFinalizeTest.Run();
+    }
+
+    private class ReRegisterForFinalizeTest
+    {
+        public static void Run()
+        {
+            TestObject.Finalized = false;
+            CreateObject();
+
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+
+            Assert.True(TestObject.Finalized);
+        }
+
+        private static void CreateObject()
+        {
+            using (var obj = new TestObject())
+            {
+                GC.SuppressFinalize(obj);
+            }
+        }
+
+        private class TestObject : IDisposable
+        {
+            public static bool Finalized { get; set; }
+
+            ~TestObject()
+            {
+                Finalized = true;
+            }
+
+            public void Dispose()
+            {
+                GC.ReRegisterForFinalize(this);
+            }
+        }
+    }
+
+    [Fact]
+    public static void GetTotalMemoryTest_ForceCollection()
+    {
+        long garbageSize = MakeSomeGarbage(1024 * 1024);
+        long garbageCollected = GC.GetTotalMemory(false) - GC.GetTotalMemory(true);
+        Assert.True(garbageCollected >= garbageSize - 10000); // 10000 bytes threshold to stabilize the test
```

Comment:
Ah I see. So even putting it in its own collection, it would still be run in parallel with other collections (aka the rest of the tests) so it wouldn't actually help.